File: frontend/src/types/wardrobe.ts
/**
 * TypeScript types for Wardrobe features.
 */

export interface ClothingItem {
  id: string;
  user_id: string;
  image_url: string;
  name: string;
  clothing_type: string;
  category: string;
  color: string;
  pattern: string | null;
  material: string | null;
  season: string | null;
  brand: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface ClothingItemCreate {
  name: string;
  clothing_type: string;
  category: string;
  color: string;
  pattern?: string;
  material?: string;
  season?: string;
  brand?: string;
  notes?: string;
}

export interface ClothingItemUpdate {
  name?: string;
  clothing_type?: string;
  category?: string;
  color?: string;
  pattern?: string;
  material?: string;
  season?: string;
  brand?: string;
  notes?: string;
}

export interface PaginationMeta {
  page: number;
  page_size: number;
  total_items: number;
  total_pages: number;
}

export interface PaginatedResponse<T> {
  success: boolean;
  data: T[];
  pagination: PaginationMeta;
}

export interface WardrobeListParams {
  page?: number;
  page_size?: number;
  search?: string;
  category?: string;
  color?: string;
  season?: string;
  clothing_type?: string;
  sort_by?: string;
  sort_order?: "asc" | "desc";
}

export const CATEGORIES = ["TOPWEAR", "BOTTOMWEAR", "FOOTWEAR", "OUTERWEAR", "ACCESSORY"] as const;
export const SEASONS = ["SUMMER", "WINTER", "SPRING", "AUTUMN", "ALL_SEASON"] as const;

File: backend/app/schemas/wardrobe.py
"""
Pydantic schemas for wardrobe operations.
"""

import uuid
from datetime import datetime, date
from pydantic import BaseModel, Field, ConfigDict


class ClothingItemCreate(BaseModel):
    """Schema for creating a clothing item (metadata sent alongside the image upload)."""
    name: str = Field(..., max_length=255)
    clothing_type: str = Field(..., max_length=100)
    category: str = Field(..., max_length=50)
    color: str = Field(..., max_length=100)
    pattern: str | None = Field(None, max_length=100)
    material: str | None = Field(None, max_length=100)
    season: str | None = Field(None, max_length=50)
    brand: str | None = Field(None, max_length=100)
    notes: str | None = None
    ai_confidence: int | None = Field(None, ge=0, le=100)
    ai_generated: bool = False
    purchase_price: float | None = None
    purchase_date: date | None = None


class ClothingItemRead(BaseModel):
    """Schema for reading a clothing item."""
    id: uuid.UUID
    user_id: uuid.UUID
    image_url: str | None = None
    name: str
    clothing_type: str
    category: str
    color: str
    pattern: str | None = None
    material: str | None = None
    season: str | None = None
    brand: str | None = None
    notes: str | None = None
    ai_confidence: int | None = None
    ai_generated: bool
    purchase_price: float | None = None
    purchase_date: date | None = None
    worn_count: int
    last_worn_at: datetime | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ClothingItemUpdate(BaseModel):
    """Schema for updating a clothing item's metadata (all fields optional)."""
    name: str | None = Field(None, max_length=255)
    clothing_type: str | None = Field(None, max_length=100)
    category: str | None = Field(None, max_length=50)
    color: str | None = Field(None, max_length=100)
    pattern: str | None = Field(None, max_length=100)
    material: str | None = Field(None, max_length=100)
    season: str | None = Field(None, max_length=50)
    brand: str | None = Field(None, max_length=100)
    notes: str | None = None
    purchase_price: float | None = None
    purchase_date: date | None = None


class PaginationMeta(BaseModel):
    """Pagination metadata included in list responses."""
    page: int
    page_size: int
    total_items: int
    total_pages: int


class ClothingItemListResponse(BaseModel):
    """Paginated list response for clothing items."""
    success: bool = True
    data: list[ClothingItemRead]
    pagination: PaginationMeta
